### ISSUE TITLE
Feature Request: Location Filter

### ISSUE THREAD

**Issue by Wensupu-Yang** [Pingid/lectures.london#20]
Thank you for creating this handy tool. 

I've noticed that many of the talks are actually for Oxford and Cambridge. While I'm happy they're included in the collection, would you consider adding a location filter for 'London Only' events?

**Comment by carolcarriazo** [comment]
Agreed - this would be great. Equally, having a column for the host name (e.g. The Royal Society, Birkbeck) would be super useful for context.

@Pingid By the way, I'd love to contribute myself to the codebase to help you out, but I noticed you excluded `lectures.json` in the `.gitgnore`. I'm sure this was intentional but let me know if you ever want some extra hands. It's a really great tool to have :)

**Comment by Pingid** [comment]
Sorry I haven’t gotten to this sooner. You can actually search for names of institutions, but I agree it makes sense to support a way to filter out non-London events. I’d imagine integrating this with the search bar somehow. I’m also not against adding a dedicated filter for institutions, since it’s clearly not obvious that you can currently do this only through search. I’m away until mid July but will try carve out some time then.

Hey @carolcarriazo 

The lectures.json file is generated by running pnpm collect. Sorry it’s not better documented. Feel free to give implementing it a try.

The code is a bit more complicated than it probably needs to be, mostly because I built it when Astro didn’t support client-side routing, and I wanted to keep things lightweight without using something like React Router. Aside from that, I hope the rest isn’t too bad.
